runtime/src/epoch_rewards_hasher.rs

```diff
@@ -55,9 +63,27 @@ pub(crate) fn hash_rewards_into_partitions(
         let partition_index = hasher
             .clone()
             .hash_address_to_partition(&reward.stake_pubkey);
-        result[partition_index].push(reward);
+        lamports[partition_index] =
+            lamports[partition_index].saturating_add(reward.stake_reward_info.lamports as u64);
+
+        if let Some(hasher) = hashers.get_mut(partition_index) {
+            hasher.hashv(&[
```

Comment:
Looks like this introduces a dependency for the ordering of the rewards in the partition for consensus.
The partition hash computation depends on the order of the rewards in the partition. If the order of rewards is different between different implementations, the consensus will be broken. 

Before this PR, the consensus doesn't require that the fixed order of rewards in the partition. As long as the partition contains the same set of rewards (regardless of the order), the consensus will still be reached.

Reply:
Yes, that is true. @t-nelson , curious to hear your thoughts on this question.

Reply:
hmm... right. i guess there are two options; 1) punt, 2) defer the sort+hash+accumulate until the block associated with that partition.

@HaoranYi did you happen to benchmark sort vs not?

Reply:
@t-nelson My first implementation is sorting. I remember sort 1M accounts by pubkey is very expensive. And you propose the hash + partition, which is the current implementation. :)

Reply:
A rough estimation of the storage overhead for the list given 1M stake accounts.
1M/4k = 250 Partitions. 
250 * 32 (hash) + 250 * 8 (lamport) = 1M bytes

It seems that the cost of the storage is not trivial.

Reply:
My vote at this point is to punt on calculating and storing the hash.

Reply:
documenting my agreement with punting. not worth the trouble

Reply:
Is that a :shipit: ✅ ?